src/test/java/uk/ac/ebi/ega/egafuse/service/EgaFileServiceTest.java

```diff
@@ -0,0 +1,168 @@
+        interceptor = new MockInterceptor(UNORDERED);
+        client = new OkHttpClient.Builder().addInterceptor(interceptor).build();
+        egaFileService = new EgaFileService(client, APP_URL, CHUNK_SIZE, PREFETCH, token, cache);
+    }
+
+    @Test
+    public void testGetFiles() throws JsonProcessingException {
+        EgaDirectory userDirectory = new EgaDirectory("EGAD00001", null, null);
+        List<File> files = new ArrayList<>();
+        File file = new File();
+        file.setFileId("EGAF00001");
+        file.setFileName("test1.cip");
+        file.setDisplayFileName("test1");
+        file.setFileSize(100l);
+        files.add(file);
+        interceptor.addRule()
+                .get(APP_URL.concat("/metadata/datasets/").concat(userDirectory.getName()).concat("/files"))
+                .respond(objectMapper.writeValueAsString(files));
+
+        List<EgaFile> userFile = egaFileService.getFiles(userDirectory);
+        File responseFile = userFile.get(0).getFile();
+        assertEquals(file.getFileId(), responseFile.getFileId());
+        assertEquals(file.getFileSize() - 16, responseFile.getFileSize());
+        assertEquals(file.getFileName(), responseFile.getFileName());
+
+    }
+
+    @Test
+    public void testGetFiles_Cip_And_Gpg() throws JsonProcessingException {
+        EgaDirectory userDirecory = new EgaDirectory("EGAD00001", null, null);
+        List<File> files = new ArrayList<>();
+        File file1 = new File();
+        file1.setFileName("/test1.cip");
+        file1.setDisplayFileName("test1");
+        File file2 = new File();
+        file2.setFileName("/test2.gpg");
+        file2.setDisplayFileName("test2");
+        files.add(file1);
+        files.add(file2);
+
+        interceptor.addRule().get(APP_URL.concat("/metadata/datasets/").concat(userDirecory.getName()).concat("/files"))
+                .respond(objectMapper.writeValueAsString(files));
+
+        List<EgaFile> userFile = egaFileService.getFiles(userDirecory);
+        assertEquals(1, userFile.size());
+    }
+
+    @Test
+    public void testGetFilesEmpty() {
+        EgaDirectory userDirecory = new EgaDirectory("EGAD00001", null, null);
+        interceptor.addRule().get(APP_URL.concat("/metadata/datasets/").concat(userDirecory.getName()).concat("/files"))
+                .respond(500);
+
+        List<EgaFile> userFile = egaFileService.getFiles(userDirecory);
+        assertTrue(userFile.isEmpty());
+    }
+
+    @Test
+    public void testFillBufferCurrentChunk() throws JsonProcessingException, InterruptedException, ExecutionException {
+        CompletableFuture<byte[]> future = mock(CompletableFuture.class);
+        when(cache.get(any())).thenReturn(future);
+        when(future.get()).thenReturn(new byte[] {});
+        long bytesToRead = 10l;
+        int chunksize = egaFileService.fillBufferCurrentChunk(pointer, "fileId", 100l, bytesToRead, 0l);
+        assertEquals(bytesToRead, chunksize);
```

Comment:
It seems wrong that it should pass when you configured the mock to not actually return any data.

Reply:
this method returns the same bytesToRead we pass into it so no need to mock it.